pkgs/tools/networking/tinystatus/default.nix

```
@@ -0,0 +1,45 @@
+{ lib
+, stdenvNoCC
+, makeWrapper
+, netcat
+, curl
+, unixtools
+, coreutils
+, mktemp
+, findutils
+, gnugrep
+, fetchFromGitHub
+,
+}:
+stdenvNoCC.mkDerivation rec {
+  postFetch = ''
+    mkdir -p $out/bin
+    tar xf $downloadedFile --strip 1
+    patchShebangs tinystatus
+    mv tinystatus $out/bin/tinystatus
+  '';
+  buildInputs = [ makeWrapper ];
+  pname = "tinystatus";
+  version = "unstable-2021-07-09";
+  src = fetchFromGitHub {
+    owner = "bderenzo";
+    repo = "tinystatus";
+    rev = "fc128adf240261ac99ea3e3be8d65a92eda52a73";
+    sha256 = "sha256-FvQwibm6F10l9/U3RnNTGu+C2JjHOwbv62VxXAfI7/s=";
+  };
+
+  installPhase = ''
+    runHook preInstall
+    install -Dm755 tinystatus $out/bin/tinystatus
+    wrapProgram $out/bin/tinystatus \
+      --set PATH "${lib.makeBinPath [stdenvNoCC curl netcat unixtools.ping coreutils mktemp findutils gnugrep]}"
```

```suggestion
      --set PATH "${lib.makeBinPath [ curl netcat unixtools.ping ]}"
```

should at least drop `stdenvNoCC` from this. I think I'd only include `curl` `netcat` and `ping` personally unless the script specifically uses gnu grep exclusive features.

---

The script fails to execute unless you have all of these.
```
❯ ./result/bin/tinystatus
/nix/store/hsgkck9biyv3idjgnpbc1ahbf3r1zzgx-tinystatus-unstable-2021-07-09/bin/.tinystatus-wrapped: line 6: mktemp: command not found
/nix/store/hsgkck9biyv3idjgnpbc1ahbf3r1zzgx-tinystatus-unstable-2021-07-09/bin/.tinystatus-wrapped: line 59: mkdir: command not found
nix/store/aw5bx76sgrgq4ryyfi2g3k272pg434zw-tinystatus-unstable-2021-07-09/bin/.tinystatus-wrapped: line 89: grep: command not found
/nix/store/aw5bx76sgrgq4ryyfi2g3k272pg434zw-tinystatus-unstable-2021-07-09/bin/.tinystatus-wrapped: line 90: [: : integer expression expe
```

Although, yes, the stdenvNoCC was left in by mistake.

---

with `--set` it fails but with `--prefix` I'd imagine it's fine

---

@06kellyjac Sure, but `--prefix` would allow impurities to enter. Therefore it would work on some people's systems, and some not, depending on whether `findutils` is installed or not.